.github/workflows/test.yml

```diff
@@ -12,7 +12,7 @@ jobs:
         with:
           python-version: ${{ matrix.python }}
       - name: Install dependencies
-        run: pip install pytest
+        run: pip install pytest pytest-mock requests
```

Comment:
`requests` theoretically shouldn't be needed for tests, but it's easier to import it here rather than refactor everything that calls it. If we want to make it very sure that no tests will reach out to any external APIs, then I can write up a mock requests module.